Src/sym.c

```diff
@@ -337,7 +337,7 @@ setallxu(Lextok *n, int t)
 
 	for (fp = n; fp; fp = fp->rgt)
 	for (tl = fp->lft; tl; tl = tl->rgt)
-	{	if (tl->sym->type == STRUCT)
+	{	if (tl->sym->type == STRUCT || tl->sym->type == UNION)
```

Comment:
there's a lot `if (STRUCT || UNION)`. I'd extract `is_record(type)` and use it everywhere (as long as promela offers grouping name for STRUCT and UNION)

Reply:
This is "user type" or "typedef", but I think this is less invasive given the rest of the codebase

Reply:
Impact on the existing code would be the same (all if's changed), but at least future modification would be simplified.

Reply:
Would it? Where do you recommend to define this function? Directly in spin.h (with no spin.c)?

Reply:
I don't know the code base, I just see code duplication and I dream about removing it :)